api/data_refinery_api/views.py

```diff
@@ -1235,26 +1236,55 @@ def get_object(self):
 # Compendia
 ###
 
-class CompendiaDetail(APIView):
+class CompendiumResultList(generics.ListAPIView):
     """
-    A very simple modified ComputedFile endpoint which only shows Compendia results.
+    List all CompendiaResults with filtering.
     """
+    model = CompendiumResult
+    queryset = CompendiumResult.objects.all()
+    filter_backends = (DjangoFilterBackend, filters.OrderingFilter,)
+    filterset_fields = ['primary_organism__name', 'compendium_version', 'quant_sf_only']
+    ordering_fields = ('primary_organism__name', 'compendium_version', 'id')
+    ordering = ('-primary_organism__name',)
+    # we need to consider filter compendium type before the grouping
+    def get_queryset(self):
+        version = self.request.query_params.get('compendium_version', False)
+        if version and version is 'latest':
+            version_filter = Q(primary_organism=OuterRef('primary_organism'),
+                               quant_sf_only=OuterRef('quant_sf_only'))
+            latest_version = CompendiumResult.objects.filter(version_filter)\
+                                                     .order_by('-compendium_version')\
+                                                     .values('compendium_version')
+            return CompendiumResult.objects.annotate(
+                latest_version=Subquery(latest_version[:1])
+            ).filter(compendium_version=F('latest_version'))
 
-    @swagger_auto_schema(deprecated=True)
-    def get(self, request, version, format=None):
+        return CompendiumResult.objects.all()
 
-        computed_files = ComputedFile.objects.filter(is_compendia=True, is_public=True, is_qn_target=False).order_by('-created_at')
-
-        token_id = self.request.META.get('HTTP_API_KEY', None)
 
+    def get_serializer_class(self):
         try:
+            token_id = self.request.META.get('HTTP_API_KEY', None)
             token = APIToken.objects.get(id=token_id, is_activated=True)
-            serializer = CompendiaWithUrlSerializer(computed_files, many=True)
+            serializer_class = CompendiumResultWithUrlSerializer
         except Exception: # General APIToken.DoesNotExist or django.core.exceptions.ValidationError
-            serializer = CompendiaSerializer(computed_files, many=True)
+            serializer_class = CompendiumResultSerializer
```

Comment:
I think we need to return `serializer_class`

Reply:
👍